### ISSUE TITLE
Handling for rerun/overrides

### ISSUE THREAD

**Issue by piotrpMSFT** [dotnet/templating#262]
Moved from https://github.com/dotnet/cli/issues/5587 on behalf of @peterblazejewicz
-----

## Steps to reproduce
- create project:
```bash
dotnet new3 console
Content generation time: 43.8456 ms
The template "Console Application" created successfully.
```

- change directory and forget about content.
- got to the same directory and run any of `new3` actions

## Expected  behavior
Handle rerun of the command on existing content, any of:
- content is overriden or there is an error about existing contetn
- there is an option to override or skip
- quit command to protect content

## Actual behavior
Code generation `fails`(?):
```bash
dotnet new3 console
Templates      Short Name      Language      Tags
-------------------------------------------------

Examples:
    dotnet new3 nugetconfig
    dotnet new3 console
    dotnet new3 --help
```
or:
```bash
dotnet new3 web
Templates      Short Name      Language      Tags
-------------------------------------------------

Examples:
    dotnet new3 mvc --auth None --framework 1.0
    dotnet new3 nugetconfig
    dotnet new3 --help
```
It looks like something broken without specific reason for the user. 
Why the `dotnet new3 nuget` works at the some directory?

```bash
dotnet new3 nuget
Content generation time: 41.2894 ms
The template "Nuget Config" created successfully.
```

## Environment data
`dotnet --info` output:
```
dotnet --info
.NET Command Line Tools (1.0.0-rc4-004597)

Product Information:
 Version:            1.0.0-rc4-004597
 Commit SHA-1 hash:  0a62481cc0

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/1.0.0-rc4-004597
````
Thanks!

**Comment by mlorbetske** [comment]
Hi @peterblazejewicz, this behavior comes from the new support for project vs item templates. If there is not a project at or above the current directory, project templates are shown - if there is, item templates are shown. Forcibly creating a project template over an existing template is supported via the `-all` switch.

That said, we should probably find a way to make this more clear. Tagging @sayedihashimi for ideas on how we should communicate this context switch & options available.

**Comment by peterblazejewicz** [comment]
Let me rephrase the subject: `There is no difference between rerun/override on existing project and handling invalid action`:

Consider: `dotnet new3` command:

```bash
dotnet new3 foo
Templates      Short Name      Language      Tags
-------------------------------------------------

Examples:
    dotnet new3 mvc --auth None --framework 1.0
    dotnet new3 classlib
    dotnet new3 --help
```

The `foo` is invalid action name, but the output is the same as one shown when I rerun correct action on already created content. At least on the version I've installed and use:

```
dotnet --info
.NET Command Line Tools (1.0.0-rc4-004597)

Product Information:
 Version:            1.0.0-rc4-004597
 Commit SHA-1 hash:  0a62481cc0

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/1.0.0-rc4-004597
```

Thanks!

**Comment by mlorbetske** [comment]
Implicit context switching has been removed for the next release after RTW - so the behavior for any supplied name will be consistent. In cases where one or more existing files would be overwritten, a message displaying the things that would happen is displayed along with an instruction to re-run the command with the `--force` parameter specified to accept the changes to existing files
